code/modules/asset_cache/asset_list_items.dm

```diff
@@ -424,7 +424,11 @@
 	else
 		icon_file = initial(item.icon)
 	var/icon_state = initial(item.icon_state)
-
+	if(ispath(item, /obj/item))
+		var/obj/item/fake_item = item
+		if(initial(fake_item.vendor_icon_preview))
+			icon_state = initial(fake_item.vendor_icon_preview)
+			icon_file = 'icons/obj/vendor_item_icons.dmi'
```

Comment:
I feel like it would make more sense just to pull it from the item's original icon file so its vendor icon is in the same place as the original icon's sprite